### DISCUSSION TITLE
From CAD to VABS input file

### DISCUSSION THREAD

**Discussion by Si1verBe11** [wenbinyugroup/.github#10] [VABS Related Discussions] [answered]
Hi,
I would like to generate a blade cross section geometry in CAD tool (CATIA, Solidworks, etc.) and use it to generate a VABS input file.
Is it possible to do this with the [Ansys-VABS GUI](https://cdmhub.org/resources/991/about) if I have an igs or stp file?
I could not find a way to import external geometry in the manual or example files for the Ansys-VABS GUI.

Most hopefully, I want to create a VABS input file directly from CATIA.
On AnalySwift's VABS introduction page, I saw that VABS integrates directly with CATIA.
I would appreciate any guidance on how this is possible.

**Comment by wenbinyu-PU** [comment, marked as answer]
The best way will be just generating VABS inputs from CATIA. Please contact your CATIA Expert for more details. We will also work with CATIA to add simple instructions into VABS manual.

**Reply by Si1verBe11** [reply]
I have additional questions.
Is it possible to perform meshing in Ansys Mechanical and then use the Ansys-VABS GUI to create a VABS input file on the Ansys Mechanical APDL?
Also, I am having trouble getting the Ansys-VABS GUI to install. Even I replaced menulist###.ans with menulist242.ans in the Ansys-VABS GUI folder (My current Ansys version is 2024 R2) Prevabs menu was not on the Ansys Mechanical APDL menu.

**Reply by wenbinyu-PU** [reply]
The Ansys-VABS GUI was developed by a student about 10 years ago. It is used to build a cross-section model layer by layer using APDL, then generate inputs for VABS. At this time, it is not possible to directly convert Ansys mechanical mesh to an input for VABS. 

Generate inputs from CATIA will be the best approach. Or you can use PreVABS to directly generate the geometry, mesh and VABS input files. If you really need to use Ansys, can you describe in more detailed about your user case why Ansys must be used? We might can add it to our future development.

**Comment by rhig2000** [comment]
CATIA FEM export options include:
- ABAQUS Input file (*.inp)
- NASTRAN Bulk Data (direct conversion) (*.nas,*.bdf)
- SIMPACK Fbi File (*.fbi)

Are any of these compatible with VABS? Does a conversion utility exist?

**Reply by wenbinyu-PU** [reply]
Yes, there is a utility existing for CATIA. However, it is not the default installation. Please contact your CATIA support regarding this.

**Reply by rhig2000** [reply]
Thanks, Wenbin! I am working with CATIA support on VABS export. However, it requires composite part design (CPD) which is typically available (for blade design development) later in the design cycle, when design maturity is higher. Whereas early design development properties are based on 2D cross-section structural geometry constructed without CPD. For this approach, CATIA offers good meshing functionality. If only we can import this 2D mesh into VABS we'll all set!

**Reply by wenbinyu-PU** [reply]
If you don't have a detailed composite model yet, what is the main difficulty preventing you using PreVABS to build the cross-section and 2D mesh? It is supposed to be very simple to use.

**Reply by rhig2000** [reply]
Lack of experience with and understanding of this PreVABS capability is a factor, certainly, but also, early 2D cross-sections are used for more than deriving structural properties. They need sufficient detail to express the manufacturing approach. Blades structures may involve bonded assembly of pre-cured laminates and other details such as foam or honeycomb core and electroformed nickel leading edge abrasion strips. Furthermore, CAD software such as CATIA can be parameterized to facilitate generating several similar cross-sections at different radial stations. Finally, these native geometries are then used to guide detail composite design development. Assuming this workflow is still necessary, we would prefer to leverage the existing CAD geometry rather than re-constructing it in another tool where discrepancies could arise.
With that in mind, I'm open to seeing what PreVABS could do. But from what I've seen, ramps from thick to thin segments are not very well defined and I'm not sure if it could handle an oval spar inside an airfoil with a variable thickness LE abrasion srtip. Do you have some examples that demonstrate its capabilities?

**Reply by wenbinyu-PU** [reply]
I will ask @unitasium to answer specific question related to PreVABS (ramps from thick to thin segments are not very well defined and I'm not sure if it could handle an oval spar inside an airfoil with a variable thickness LE abrasion srtip). PreVABS has no problem generating several similar sections at different radial stations. 

Will CATIA support is willing to adopt what they already have to what you need? It looks to me a simple matter since they already have it for CPD. 

The last option will be for us to develop such a utility converting from CATIA to VABS along with your folks.

**Reply by rhig2000** [reply]
Sounds good. Good point regarding CATIA support.

**Reply by rhig2000** [reply]
I reached out to DS

**Reply by unitasium** [reply]
@rhig2000 We recently developed a python package called `sgio` to manage and operate cross-sectional data for different tools. You can find it here: https://github.com/wenbinyugroup/sgio.

One capability of this package is to convert mesh data between different formats, including Abaqus inp and VABS input. There is an example to show this. With this, you can export your model to the .inp file and then convert it to VABS input.

However, this functionality handles one specific way of creating meshed cross-sections in Abaqus at this moment. Considering that there could be multiple ways to create a model (different ways to assign domain, material section, local orientation, element type, etc.), this tool may not be immediately workable for your case. If not, you can send me an example of your case and we will update the code for better compatibility.

**Reply by rhig2000** [reply]
I ran into an issue with installing the package. Please help diagnose:

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [54 lines of output]

**Comment by rhig2000** [comment]
Thanks @unitasium, I'll check that out.

Dassault Systems was contacted regarding incorporating the VABS compatibility into the workflow. DS is open to this possibility, but full implementation may be a year away. Meanwhile, DS advised VABS should be upgraded to accept standard FEM file formats. The advantage of defining the mesh in native CATIA is that it can be associatively updated automatically with any geometry changes.
